examples/chip-tool-darwin/templates/commands.zapt

```diff
@@ -0,0 +1,368 @@
+        {{#if isArray}}
+        // {{label}} Array parsing is not supported yet
+        {{else if isStruct}}
+        // {{label}} Struct parsing is not supported yet
+        {{else if (isString type)}}
+        AddArgument("attr-value", &mValue);
+        {{else}}
+        AddArgument("attr-value", {{asTypeMinValue type}}, {{asTypeMaxValue type}}, &mValue);
+        {{/if}}
+        ModelCommand::AddArguments();
+    }
+
+    ~Write{{asUpperCamelCase parent.name}}{{asUpperCamelCase name}}()
+    {
+    }
+
+    CHIP_ERROR SendCommand(CHIPDevice * device, chip::EndpointId endpointId) override
+    {
+        ChipLogProgress(chipTool, "Sending cluster ({{asHex parent.code 8}}) WriteAttribute ({{asHex code 8}}) on endpoint %" PRIu16, endpointId);
+
+        {{#if isArray}}
+        {{else if isStruct}}
+        {{else if (isOctetString type)}}
+        AddArgument("attr-value", &mValue);
+        {{else if (isCharString type)}}
+        AddArgument("attr-value", &mValue);
+        {{else}}
+        AddArgument("attr-value", {{asTypeMinValue type}}, {{asTypeMaxValue type}}, &mValue);
+        {{/if}}
+
+        dispatch_queue_t callbackQueue = dispatch_queue_create("com.chip.command", DISPATCH_QUEUE_SERIAL);
+        CHIP{{asUpperCamelCase parent.name}} * cluster = [[CHIP{{asUpperCamelCase parent.name}} alloc] initWithDevice:device endpoint:endpointId queue:callbackQueue];
+
+        {{#if isArray}}
+        // {{label}} Array parsing is not supported yet
+        {{else if isStruct}}
+        // {{label}} Struct parsing is not supported yet
+        {{else if (isOctetString type)}}
+        {{asObjectiveCType type parent.name}} value = [[NSData alloc] initWithBytes:mValue.data() length:mValue.size()];
+        {{else if (isString type)}}
+        {{asObjectiveCType type parent.name}} value = [[NSString alloc] initWithBytes:mValue.data() length:mValue.size() encoding:NSUTF8StringEncoding];
+
+        {{else}}
+        {{asObjectiveCType type parent.name}} value = [NSNumber numberWith{{asObjectiveCNumberType "" type false}}:mValue];
+        {{/if}}
+        [cluster writeAttribute{{asUpperCamelCase name}}WithValue:value completionHandler:^(NSError * _Nullable error) {
+            CHIP_ERROR err = CHIP_NO_ERROR;
+            err = [CHIPError errorToCHIPErrorCode:error];
+
+            ChipLogError(chipTool, "{{asUpperCamelCase parent.name}} {{asUpperCamelCase name}} Error: %s", chip::ErrorStr(err));
+            SetCommandExitStatus(err);
+            }];
+        return CHIP_NO_ERROR;
+    }
+
+private:
+    {{#if isArray}}
+    {{else if isStruct}}
+    {{else if (isOctetString type)}}
+    chip::ByteSpan mValue;
+    {{else if (isCharString type)}}
+    chip::ByteSpan mValue;
+    {{else}}
+    {{chipType}} mValue;
+    {{/if}}
+};
+
+{{/unless}}
+{{/if}}
+{{#if isReportableAttribute}}
+class Report{{asUpperCamelCase parent.name}}{{asUpperCamelCase name}}: public ModelCommand
```

Comment:
How about naming things 'subscription' and similar to be more in line with matter spec?